aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1071,34 +1092,29 @@ Tensor& addmm_cpu_out(const Tensor& self, const Tensor& mat1, const Tensor& mat2
     at::NoNamesGuard guard;
     addmm_impl_cpu_(result, *b_self, mat1, mat2, beta, alpha);
   }
-  at::namedinference::propagate_names_for_addmm(result, mat1, mat2, self);
+  auto names = at::namedinference::propagate_names_for_addmm(mat1, mat2, self);
+  at::namedinference::propagate_names_if_nonempty(result, names);
   return result;
 }
 
-Tensor addmm_cpu(const Tensor& self, const Tensor& mat1, const Tensor& mat2, const Scalar& beta, const Scalar& alpha) {
-  Tensor result = at::empty({0}, self.options());
-  return addmm_cpu_out(self, mat1, mat2, beta, alpha, result);
-}
-
-Tensor &addmm_cpu_(Tensor& self, const Tensor& mat1, const Tensor& mat2, const Scalar& beta, const Scalar& alpha) {
-  TORCH_CHECK(((self.dim() == 2) && (self.sizes()[0] == mat1.sizes()[0]) && (self.sizes()[1] == mat2.sizes()[1])),
```

Comment:
@ezyang The test failed because I delete this check. But this check is only for `addmm_cpu_` (it is inplace op), how can I implement this check in `structured addmm`?